devito/function.py

```diff
@@ -631,6 +647,9 @@ class TimeFunction(Function):
 
     def __init__(self, *args, **kwargs):
         if not self._cached():
+            time_order = kwargs.get('time_order', 1)
+            self.derivatives = ["dt"] if time_order > 0 else []
+            self.derivatives += ["dt2"] if time_order > 1 else []
```

Comment:
again: why do we even need this instance attribute ``derivatives``? when do we need it?